components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -531,31 +532,78 @@ private static CallBackHandlerFactory getCallBackHandlerFactory() {
             throws IOException {
         // TODO read the URL from framework config file rather than carbon.xml
-        request.setAttribute(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus.INCOMPLETE);
-        response.sendRedirect(getRedirectURL(ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL(),
-                request));
+        sendToRetryPage(request, response, null, null);
     }
 
-
+    /**
+     * Send user to retry page during an authentication flow failure.
+     *
+     * @param request       Http servlet request.
+     * @param response      Http servlet response.
+     * @param status        Failure status.
+     * @param statusMsg     Failure status message.
+     * @throws IOException
+     * @deprecated use {@link #sendToRetryPage(HttpServletRequest, HttpServletResponse, AuthenticationContext, String, String)}.
+     */
+    @Deprecated
     public static void sendToRetryPage(HttpServletRequest request, HttpServletResponse response, String status,
                                        String statusMsg) throws IOException {
 
+        sendToRetryPage(request, response, null, status, statusMsg);
+    }
+
+    /**
+     * Send user to retry page during an authentication flow failure.
+     *
+     * @param request       Http servlet request.
+     * @param response      Http servlet response.
+     * @param context       Authentication Context.
+     * @param status        Failure status.
+     * @param statusMsg     Failure status message.
+     * @throws IOException
+     */
+    public static void sendToRetryPage(HttpServletRequest request, HttpServletResponse response,
+                                       AuthenticationContext context, String status,
+                                       String statusMsg) throws IOException{
+
         try {
             URIBuilder uriBuilder = new URIBuilder(
                     ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL());
-            uriBuilder.addParameter("status", status);
-            uriBuilder.addParameter("statusMsg", statusMsg);
+            if (status != null && statusMsg != null) {
+                uriBuilder.addParameter("status", status);
+                uriBuilder.addParameter("statusMsg", statusMsg);
+            }
             request.setAttribute(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus.INCOMPLETE);
-            response.sendRedirect(getRedirectURL(uriBuilder.build().toString(), request));
+            if (context != null) {
+                if (IdentityTenantUtil.isTenantedSessionsEnabled()) {
+                    uriBuilder.addParameter(USER_TENANT_DOMAIN_HINT, context.getUserTenantDomain());
+                }
+                uriBuilder.addParameter(REQUEST_PARAM_SP, context.getServiceProviderName());
+                uriBuilder.addParameter(TENANT_DOMAIN, context.getTenantDomain());
```

Comment:
let's add tenant domain only if ((!IdentityTenantUtil.isTenantQualifiedUrlsEnabled())
https://github.com/wso2/carbon-identity-framework/pull/3373/files#diff-c827402ed6c15cc879db866bf51cbd2c886fd806013a4c5ad06c9b47bb3a8298L584

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/3373/commits/2f9dc5c391cb14eb4538bda7de2b38da9c3c17c5